x/auth/ante/ante.go

```diff
@@ -1,423 +1,26 @@
-	}
-
-	// verify the account has enough funds to pay for fees
-	_, hasNeg := coins.SafeSub(fees)
-	if hasNeg {
-		return sdk.ErrInsufficientFunds(
-			fmt.Sprintf("insufficient funds to pay for fees; %s < %s", coins, fees),
-		).Result()
-	}
-
-	// Validate the account has enough "spendable" coins as this will cover cases
-	// such as vesting accounts.
-	spendableCoins := acc.SpendableCoins(blockTime)
-	if _, hasNeg := spendableCoins.SafeSub(fees); hasNeg {
-		return sdk.ErrInsufficientFunds(
-			fmt.Sprintf("insufficient funds to pay for fees; %s < %s", spendableCoins, fees),
-		).Result()
-	}
-
-	err := supplyKeeper.SendCoinsFromAccountToModule(ctx, acc.GetAddress(), types.FeeCollectorName, fees)
-	if err != nil {
-		return err.Result()
-	}
-
-	return sdk.Result{}
-}
-
-// EnsureSufficientMempoolFees verifies that the given transaction has supplied
-// enough fees to cover a proposer's minimum fees. A result object is returned
-// indicating success or failure.
-//
-// Contract: This should only be called during CheckTx as it cannot be part of
-// consensus.
-func EnsureSufficientMempoolFees(ctx sdk.Context, stdFee types.StdFee) sdk.Result {
-	minGasPrices := ctx.MinGasPrices()
-	if !minGasPrices.IsZero() {
-		requiredFees := make(sdk.Coins, len(minGasPrices))
-
-		// Determine the required fees by multiplying each required minimum gas
-		// price by the gas limit, where fee = ceil(minGasPrice * gasLimit).
-		glDec := sdk.NewDec(int64(stdFee.Gas))
-		for i, gp := range minGasPrices {
-			fee := gp.Amount.Mul(glDec)
-			requiredFees[i] = sdk.NewCoin(gp.Denom, fee.Ceil().RoundInt())
-		}
-
-		if !stdFee.Amount.IsAnyGTE(requiredFees) {
-			return sdk.ErrInsufficientFee(
-				fmt.Sprintf(
-					"insufficient fees; got: %q required: %q", stdFee.Amount, requiredFees,
-				),
-			).Result()
-		}
-	}
-
-	return sdk.Result{}
-}
-
-// SetGasMeter returns a new context with a gas meter set from a given context.
-func SetGasMeter(simulate bool, ctx sdk.Context, gasLimit uint64) sdk.Context {
-	// In various cases such as simulation and during the genesis block, we do not
-	// meter any gas utilization.
-	if simulate || ctx.BlockHeight() == 0 {
-		return ctx.WithGasMeter(sdk.NewInfiniteGasMeter())
-	}
-
-	return ctx.WithGasMeter(sdk.NewGasMeter(gasLimit))
-}
-
-// GetSignBytes returns a slice of bytes to sign over for a given transaction
-// and an account.
-func GetSignBytes(chainID string, stdTx types.StdTx, acc exported.Account, genesis bool) []byte {
-	var accNum uint64
-	if !genesis {
-		accNum = acc.GetAccountNumber()
-	}
-
-	return types.StdSignBytes(
-		chainID, accNum, acc.GetSequence(), stdTx.Fee, stdTx.Msgs, stdTx.Memo,
+	return sdk.ChainAnteDecorators(
+		NewSetUpContextDecorator(),
+		NewMempoolFeeDecorator(),
+		NewValidateBasicDecorator(),
+		NewValidateMemoDecorator(ak),
+		NewConsumeGasForTxSizeDecorator(ak),
+		NewSetPubKeyDecorator(ak),
+		NewValidateSigCountDecorator(ak),
```

Comment:
hmm so the order of the decorators doesn't match the order of the previous `AnteHandler`. Would it be possible to reorder it again or add a comment on the new order of the AnteDecorators?

Reply:
Correct, I tried changing the order to precisely match old code but it didn't work well (returned errors that weren't very helpful)

If u have specific suggestions on changed order, i can try them out. But i should definitely add some documentation on how the total antehandler flow works now